k8s/clients.go

```diff
@@ -71,11 +75,29 @@ func NewClients(config ClientsConfig) (*Clients, error) {
 		}
 	}
 
+	var helmClient helmclient.Interface
+	{
+		c := helmclient.Config{
+			K8sClient:  k8sClient,
+			Logger:     config.Logger,
+			RestConfig: restConfig,
+
+			TillerImage:     config.TillerImage,
+			TillerNamespace: config.TillerNamespace,
```

Comment:
Both tiller settings are optional and helmclient handles applying the default values.

Reply:
I am not sure about this in general - why do we need helmclient here? What is the motivation to include it here instead of having it in the usual wiring as before?

I am a bit afraid that it will make our import cycles a bit worse right now.

Reply:
Yeah I'm also worried about dependencies. Maybe separate package?

Reply:
It comes from here. https://github.com/giantswarm/e2etests/pull/46/files#r298913958

But yes helmclient screws up our import cycles and we need to make that better not worse. What about if I separate this to `github.com/giantswarm/e2esetup/helmclient`?

Since we need to use the helm directory for charts.

Reply:
`e2esetup/helmclient` is fine with me 👍

Reply:
Maybe we should rename `k8s` to `k8sclient` for consistency but I'm not super fussed if this is a hassle.

Reply:
Package idea is fine with me. No real blockers about naming from my side.

Reply:
Sweet I'll update this to use that package structure.

Not sure on changing k8s to k8sclient as it also has generic logic like creating CRDs and namespaces.

Reply:
> Not sure on changing k8s to k8sclient as it also has generic logic like creating CRDs and namespaces.

Yeah I forgot. That doesn't make sense then.

Reply:
Or we should move everything to `pkg`.